BGL/include/CGAL/boost/graph/Named_function_parameters.h

```diff
@@ -204,8 +204,7 @@ get_parameter(const Named_function_parameters<T, Tag, Base>& np, Query_tag tag)
 }
 
 template <typename D>
-const D&
-choose_parameter(const internal_np::Param_not_found&, const D& d)
+D choose_parameter(const internal_np::Param_not_found&, const D& d)
 {
   return d;
 }
```

Comment:
Is not that a place where we should use move-semantic and perfect forwarding, like in the new version of Boost.Parameters?

With the current implementation, the argument passed in the parameter `d` is guaranteed to be copied. Were `d` passed by value, instead of reference to const, then the move semantic could be used if `D` is heavy.

On the other hand, in CGAL BGL API, I think all our named parameters are actually either lightweight property maps, or trivial parameters (type `int`, `double`, or at worse a small struct).

Reply:
I was thinking about it some other day with the fact that you could have as default value a traits class that wouldn't like copies due to some internal members. It was quite specific, but I guess moving it would be a good solution.

In any case, this targets 4.13 so it just needs to copy here.

Reply:
It targets 4.13 but 4.13 does not prevent users from using C++11 or later. C++14 is the default since gcc 6 and MSVC 2015, for example.

But I agree with can think about that later, once we have a real use case to test with.